fieldservice/models/fsm_location.py

```diff
@@ -256,18 +256,18 @@ def action_view_contacts(self):
         for location in self:
             action = self.env.ref('contacts.action_contacts').\
                 read()[0]
-            contacts = self.get_action_views(1, 0, location)
+            locations = self.get_action_views(1, 0, location)
             action['context'] = self.env.context.copy()
+            action['context'].update({'group_by': ''})
             action['context'].update({'default_service_location_id': self.id})
-            if len(contacts) == 1:
-                action['views'] = [(self.env.ref('base.view_partner_form').id,
+            if len(locations) == 0 or len(locations) > 1:
+                action['domain'] = [('id', 'in', locations.ids)]
+            elif locations:
+                action['views'] = [(self.env.
+                                    ref('base.' +
+                                        'view_partner_form').id,
                                     'form')]
-                action['res_id'] = contacts.id
-                action['context'].update({'active_id': contacts.id})
-            else:
-                action['domain'] = [('id', 'in', contacts.ids)]
-                action['context'].update({'active_ids': contacts.ids})
-                action['context'].update({'active_id': ''})
+                action['res_id'] = locations.id
             return action
```

Comment:
@osi-scampbell I am not sure to follow you here: This is the method to view contacts from the location form view and based on the number of contacts, we display the form or list view of contacts... The variable name seems correct to me.

Reply:
@max3903 You are correct this variable is named improperly